### PR TITLE
Add explicit routes for all celery tasks, refs #1228

### PR DESCRIPTION
**Pull request by EnTeQuAk**
Defined routes:

* priority
* devhub
* images
* limited
* amo
* addons
* api
* cron
* bandwagon
* editors
* files
* crypto
* search
* reviews
* stats
* tags
* users
* zadmin

### CODE REVIEW

**Comment by andymckay** [comment]
Assigning to @jasonthomas because we'll need to make the queues for these in the celery config.

File: lib/settings_base.py [hunk 1]
```diff
@@ -1002,10 +1003,22 @@ def JINJA_CONFIG():
...
+    'devhub.tasks.resize_icon': {'queue': 'devhub'},
+    'devhub.tasks.resize_preview': {'queue': 'devhub'},
+    'devhub.tasks.send_welcome_email': {'queue': 'devhub'},
+    'devhub.tasks.submit_file': {'queue': 'devhub'},
+    'devhub.tasks.validate_file': {'queue': 'devhub'},
     'devhub.tasks.validate_file': {'queue': 'devhub'},
     'devhub.tasks.validate_file_path': {'queue': 'devhub'},
-    'devhub.tasks.handle_upload_validation_result': {'queue': 'devhub'},
-    'devhub.tasks.handle_file_validation_result': {'queue': 'devhub'},
+    'devhub.tasks.validate_file_path': {'queue': 'devhub'},
```
Comment (mstriemer) [inline lib/settings_base.py]: Looks like this one is duplicated now.

File: lib/settings_base.py [hunk 2]
```diff
@@ -1002,10 +1003,22 @@ def JINJA_CONFIG():
     # Other queues we prioritize below.
 
     # AMO Devhub.
+    'devhub.tasks.convert_purified': {'queue': 'devhub'},
+    'devhub.tasks.flag_binary': {'queue': 'devhub'},
+    'devhub.tasks.get_preview_sizes': {'queue': 'devhub'},
+    'devhub.tasks.handle_file_validation_result': {'queue': 'devhub'},
+    'devhub.tasks.handle_file_validation_result': {'queue': 'devhub'},
+    'devhub.tasks.handle_upload_validation_result': {'queue': 'devhub'},
+    'devhub.tasks.handle_upload_validation_result': {'queue': 'devhub'},
```
Comment (mstriemer) [inline lib/settings_base.py]: `devhub.tasks.handle_upload_validation_result` and `devhub.tasks.handle_file_validation_result` duplicated too.

File: lib/settings_base.py [hunk 3]
```diff
@@ -1002,10 +1003,22 @@ def JINJA_CONFIG():
...
+    'devhub.tasks.resize_icon': {'queue': 'devhub'},
+    'devhub.tasks.resize_preview': {'queue': 'devhub'},
+    'devhub.tasks.send_welcome_email': {'queue': 'devhub'},
+    'devhub.tasks.submit_file': {'queue': 'devhub'},
+    'devhub.tasks.validate_file': {'queue': 'devhub'},
```
Comment (mstriemer) [inline lib/settings_base.py]: And this one.

**Comment by kumar303** [comment]
Sweet! r+ from me. 

One thing worth noting (but not related to this patch) is that if any view imports a task like `from apps.foo.tasks import the_task` then the import path would be prefixed with `apps` and it would need to be routed differently. The correct way to do it is `from foo.tasks import...`. When grepping I don't see any offenders but it would be nice to flake8 this (or something) to prevent it.

**Comment by diox** [comment]
That's an interesting remark, we'll have to take that into account in the module refactor PR (#1081) then, since it changes all the imports.

**Comment by jasonthomas** [comment]
I didn't create a worker per queue, I want to maximize idle workers and reduce worker RAM usage. Let me know if the worker/queue arrangement makes sense or requires any changes. 

* olympia worker
 * queues = celery,default,api,files,search,tags
* olympia-cron worker
 * queues = cron,stats,review
* olympia-crypto worker
 * queues = crypto
* olympia-devhub worker
 * queues = devhub
* olympia-editors worker
 * queues = editors,zadmin
* olympia-images worker
 * queues = images
* olympia-limited worker
 * queues = limited
* olympia-priority worker
 * queues = priority,amo,addons,bandwagon,users

**Comment by EnTeQuAk** [comment]
Thanks  @mstriemer for review, fixed those duplicate. How embarrassing :sweat_smile:

**Comment by EnTeQuAk** [comment]
@jasonthomas Looks good to me I think

**Comment by jasonthomas** [comment]
Thanks for review. I've merged in the changes.

**Pull request merged**